blockchain/chainio.go

```diff
@@ -1576,7 +1576,7 @@ func (b *BlockChain) initChainState() error {
 		}
 
 		// The database bucket for the versioning information is missing.
-		if dbInfo == nil && err == nil {
```

Comment:
This is wrong.  It is entirely possible for `dbInfo` to be `nil` and the `err` to be `nil` at the same time which specifically indicates an uninitialized state.  This would break the code.